source/common/network/dns_impl.cc

```diff
@@ -104,16 +104,24 @@ void DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback(int status, i
     // ARES_ECONNREFUSED. If the PendingResolution has not been cancelled that means that the
```

Comment:
The c-ares based resolver changed a bit more because it was not originally written to accumulate responses from multiple c-ares callbacks.

Reply:
I refactored for ease of understanding but open to comments to simplify further!